dkg/pedersen/handler.go

```diff
@@ -185,144 +195,112 @@ func (h *Handler) Stream(out mino.Sender, in mino.Receiver) error {
 
 func (h *Handler) handleDecrypt(out mino.Sender, msg types.DecryptRequest,
 	from mino.Address) error {
-	dela.Logger.Trace().Msgf(
-		"%v received decrypt request from %v\n", h.me, from,
-	)
+
 	if !h.startRes.Done() {
-		return xerrors.Errorf(
-			"you must first initialize DKG. Did you " +
-				"call setup() first?",
-		)
+		return xerrors.Errorf("you must first initialize DKG. Did you call setup() first?")
 	}
 
-	// TODO: check if started before
 	h.RLock()
 	S := suite.Point().Mul(h.privShare.V, msg.K)
 	h.RUnlock()
 
 	partial := suite.Point().Sub(msg.C, S)
 
 	h.RLock()
-	decryptReply := types.NewDecryptReply(
-		// TODO: check if using the private index is the same as the public
-		// index.
-		int64(h.privShare.I),
-		partial,
-	)
+	decryptReply := types.NewDecryptReply(int64(h.privShare.I), partial)
 	h.RUnlock()
 
 	errs := out.Send(decryptReply, from)
 	err := <-errs
 	if err != nil {
-		return xerrors.Errorf(
-			"got an error while sending the decrypt "+
-				"reply: %v", err,
-		)
-	}
-	return nil
-}
-
-func (h *Handler) handleStart(ctx context.Context, out mino.Sender,
-	from mino.Address, msg types.Start, done chan struct{}) error {
-	if h.startRes.Done() {
-		dela.Logger.Warn().Msgf(
-			"%v ignored start request from %v as it is already"+
-				" started\n", h.me, from)
-		return xerrors.Errorf("dkg is already started")
-	}
-
-	if h.startRes.IsStarting() {
-		dela.Logger.Warn().Msgf(
-			"%v ignored start request from %v as it is already"+
-				" starting\n", h.me, from)
-		return xerrors.Errorf("dkg is already starting")
+		return xerrors.Errorf("got an error while sending the decrypt reply: %v", err)
 	}
 
-	h.startRes.Start()
-
-	go func() {
-		err := h.start(ctx, msg, from, out)
-		if err != nil {
-			dela.Logger.Error().Msgf(
-				"%v failed to start: %v", h.me, err)
-			h.startRes.AbortStart()
-		}
-		close(done)
-	}()
-
 	return nil
 }
 
 // start is called when the node has received its start message. Note that we
 // might have already received some deals from other nodes in the meantime. The
 // function handles the DKG creation protocol.
-func (h *Handler) start(ctx context.Context, start types.Start,
-	from mino.Address, out mino.Sender) error {
-
-	dela.Logger.Info().Msgf("%v is starting a DKG", h.me)
+func (h *Handler) start(start types.Start, deals, resps *list.List, from mino.Address,
+	out mino.Sender) error {
 
-	participants := start.GetAddresses()
-
-	if len(participants) != len(start.GetPublicKeys()) {
-		return xerrors.Errorf(
-			"there should be as many participants as "+
-				"pubKey: %d != %d", len(start.GetAddresses()),
-			len(start.GetPublicKeys()),
-		)
+	if len(start.GetAddresses()) != len(start.GetPublicKeys()) {
+		return xerrors.Errorf("there should be as many participants as "+
+			"pubKey: %d != %d", len(start.GetAddresses()), len(start.GetPublicKeys()))
 	}
-	n := len(participants)
-	ctx, cancel := context.WithTimeout(ctx, time.Duration(n*n)*time.Second)
-	defer cancel()
```

Comment:
shouldn't we time out after an arbitrary long amount of time, if the DKG is completely stalled ? 
It seems brittle not to have a max timeout for the DKG, to at least return "this failed"